cmd/releaseagent/internal/coordinator/step.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package coordinator
+
+import (
+	"compress/zlib"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+)
+
+const NoTimeout = time.Duration(0)
+
+type StepFunc func(ctx context.Context) error
+
+// Step represents a step in the release. Just enough information is represented in Step to allow
+// status to be reported, otherwise state is internal to Impl.
+type Step struct {
+	// Name is the name of the step. It must be unique within the release step graph.
+	Name string
+	// Timeout defines the deadline that should be set up for the ctx passed to Impl.
+	// If NoTimeout (zero), no deadline is set.
+	Timeout time.Duration
+	// Impl is the implementation of the step. It is executed when the step is run.
+	// It is run in its own goroutine and may block on network calls, retries, etc.
+	// It shouldn't wait for another step to complete: this should be done using DependsOn.
+	Impl StepFunc
```

Comment:
nit: I prefer the name `Func`, it is clearer than `Impl` and it fits better with the type that define it (`StepFunc`).